### PR TITLE
[WIP] Added DHCPv4 server

### PR DESCRIPTION
**Pull request by insomniacslk**
Added DHCPv4 server code and tests.

Still WIP. Tests will fail: requires root access; and port 0 and raw sockets.

### CODE REVIEW

**Comment by insomniacslk** [comment]
To simplify this PR I need to implement `NewOfferFromDiscovery` and `NewAckFromRequest`

**Comment by insomniacslk** [comment]
The server tests must go in their own integration tests package

File: dhcpv4/server_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	if m.Opcode() != OpcodeBootRequest {
+		log.Printf("Not a BootRequest!")
+		return
+	}
+	response, err := New()
+	if err != nil {
+		log.Printf("New DHCPv4 packet creation failed: %v", err)
+		return
+	}
+	// TODO implement NewOfferFromDiscovery and NewAckFromRequest,
```
Comment (pmazzini) [inline dhcpv4/server_test.go]: How about using `NewReplyFromRequest` here?

Comment (insomniacslk) [reply dhcpv4/server_test.go]: you're damn right, sir!

File: dhcpv4/server_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+func TestServerActivateAndServe(t *testing.T) {
+	c, s := setUpClientAndServer(DORAHandler)
+	defer s.Close()
+
+	lo, err := getLoopbackInterface()
+	require.NoError(t, err)
+
+	hwaddr := []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06}
+	discover, err := NewDiscovery(hwaddr)
```
Comment (pmazzini) [inline dhcpv4/server_test.go]: Do we need to create a discover here? How about passing nil to the `Exchange`?

Comment (insomniacslk) [reply dhcpv4/server_test.go]: good point. My initial diff contained a refactor of `Exchange` into `Discover` and `Request`, but I removed it for a future diff. I still need to set the transaction ID and some other stuff, but I can do that via modifiers

**Comment by insomniacslk** [comment]
> I didn't find anything to comment on. Approving to unblock you.

Thanks! I'll try to get coverage to be green before merging :)

**Pull request merged**